libgraph/include/katana/analytics/kSsspImplementationBase.h

```diff
@@ -0,0 +1,296 @@
+/*
```

Comment:
I tried doing inheritance for BfsSsspImplementationBase, shared structs and functions are not easy to reuse due to the added Path variable

Reply:
daryan is right. doing the inheritance thing is not going to help much here sadly.

Reply:
What's the reason for that? Are files completely completely different? or is there some other complexity?

Reply:
The key difference between kSssp and BfsSssp is that kSssp has an added Path variable for the structs and functions. Because of this, functions and structs need to be re-implemented to support this Path variable, which also means the other variables need to be re-implemented for those functions to compile, which means we it makes no difference to reuse BfsSssp unless we want to add a Path option to it (but I'm not sure how well that would work)

Reply:
yes, daryan is right